project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/TransportDataFixture.php

```diff
@@ -44,46 +71,46 @@ public function __construct(
-        ];
+
+        foreach ($this->domain->getAllLocales() as $locale) {
+            $transportData->name[$locale] = t('Czech post', [], 'dataFixtures', $locale);
+        }
+
+        foreach ($this->domain->getAllIncludingDomainConfigsWithoutDataCreated() as $domain) {
+            $currency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domain->getId());
+            $transportData->pricesByCurrencyId[$currency->getId()] = Money::create((string)(99.95 / $currency->getExchangeRate()));
+        }
+
         $transportData->vat = $this->getReference(VatDataFixture::VAT_HIGH);
         $this->createTransport(self::TRANSPORT_CZECH_POST, $transportData);
 
         $transportData = $this->transportDataFactory->create();
-        $transportData->name = [
-            'cs' => 'PPL',
-            'en' => 'PPL',
-        ];
-        $transportData->pricesByCurrencyId = [
-            $this->getReference(CurrencyDataFixture::CURRENCY_CZK)->getId() => Money::create('199.95'),
-            $this->getReference(CurrencyDataFixture::CURRENCY_EUR)->getId() => Money::create('6.95'),
-        ];
+
+        foreach ($this->domain->getAllLocales() as $locale) {
+            $transportData->name[$locale] = t('PPL', [], 'dataFixtures', $locale);
+        }
+
+        $this->setPriceForAllCurrencies($transportData, Money::create('199.95'));
+
         $transportData->vat = $this->getReference(VatDataFixture::VAT_HIGH);
         $this->createTransport(self::TRANSPORT_PPL, $transportData);
 
         $transportData = $this->transportDataFactory->create();
-        $transportData->name = [
-            'cs' => 'Osobní převzetí',
-            'en' => 'Personal collection',
-        ];
-        $transportData->pricesByCurrencyId = [
-            $this->getReference(CurrencyDataFixture::CURRENCY_CZK)->getId() => Money::zero(),
-            $this->getReference(CurrencyDataFixture::CURRENCY_EUR)->getId() => Money::zero(),
-        ];
-        $transportData->description = [
-            'cs' => 'Uvítá Vás milý personál!',
-            'en' => 'You will be welcomed by friendly staff!',
-        ];
-        $transportData->instructions = [
-            'cs' => 'Těšíme se na Vaši návštěvu.',
-            'en' => 'We are looking forward to your visit.',
-        ];
+
+        foreach ($this->domain->getAllLocales() as $locale) {
+            $transportData->name[$locale] = t('Personal collection', [], 'dataFixtures', $locale);
+        }
+
+        $this->setPriceForAllCurrencies($transportData, Money::zero());
+
+        foreach ($this->domain->getAllLocales() as $locale) {
+            $transportData->description[$locale] = t('You will be welcomed by friendly staff!', [], 'dataFixtures', $locale);
+        }
+
+        foreach ($this->domain->getAllLocales() as $locale) {
+            $transportData->instructions[$locale] = t('We are looking forward to your visit.', [], 'dataFixtures', $locale);
+        }
```

Comment:
again... DRY violation with the foreaches. can you please join them somehow?

Reply:
fixed in 38a656483